Cypress Tests Failure

Cypress Tests are failing in the github actions and It seems to be blocking most of the PRs right now in repository.

There has been some initial discussion at https://github.com/publiclab/leaflet-environmental-layers/pull/529

Specifically Purple Air Marker Layer and Owm Layer tests are failing. 
![image](https://user-images.githubusercontent.com/47106543/123113125-ebe99b80-d45b-11eb-999c-a09f97451122.png)

Maybe related to #360 ?

Noting https://github.com/publiclab/leaflet-environmental-layers/pull/529#issuecomment-870934595, the tests initially passed when the actions were setup

I am taking a look at this... It's my first time working with the `leaflet-environmental-layers` library, and I'm not familiar with the Cypress testing library either. So basically, there's a lot I don't yet know about this! 😅 

Maybe it would be helpful to compile the testing errors from CI:
<img width="828" alt="Screen Shot 2021-07-12 at 1 46 22 PM" src="https://user-images.githubusercontent.com/4361605/125353550-91838280-e317-11eb-841f-e5e28bb23c97.png">
<img width="891" alt="Screen Shot 2021-07-12 at 1 45 52 PM" src="https://user-images.githubusercontent.com/4361605/125353571-95170980-e317-11eb-9b61-63777fe2a5f1.png">

Here they are in raw text format, to make searching easier:
```
   1) OpenWeatherMap layer
       adds markers on click:
     AssertionError: Timed out retrying: Expected to find element: `#map-city label`, but never found it.
      at Context.eval (http://localhost:8080/__cypress/tests?p=cypress/integration/owm.spec.js:10:10)

  2) OpenWeatherMap layer
       loads correct owmloading.gif with config option set:
     ReferenceError: LEL is not defined
      at Context.eval (http://localhost:8080/__cypress/tests?p=cypress/integration/owm.spec.js:21:14)
```

```
Purple layer - markers
    ✓ adds markers on click (1158ms)
    ✓ should have the layer name added to the hash
    ✓ has default markers in default mode (38ms)
    1) has circle markers in minimal mode
    2) shows popup
    ✓ removes markers from the map and the layer name from the hash when clicked again (157ms)


  4 passing (10s)
  2 failing

  1) Purple layer - markers
       has circle markers in minimal mode:

      AssertionError: Timed out retrying: Too many elements found. Found '3', expected '2'.
      + expected - actual

      -3
      +2
      
      at Context.eval (http://localhost:8080/__cypress/tests?p=cypress/integration/purpleAirMarker.spec.js:33:56)

  2) Purple layer - markers
       shows popup:
     AssertionError: Timed out retrying: expected '<div.leaflet-popup-content>' to contain 'MandMnorth40'
      at Context.eval (http://localhost:8080/__cypress/tests?p=cypress/integration/purpleAirMarker.spec.js:45:38)
```

So @daemon1024 i think here, let's go through these one by one in a PR. Let's start with an easy one -- this line references a specific sensor that's no longer present, we assume:

https://github.com/publiclab/leaflet-environmental-layers/blob/2f3723a12e70009120500f3b7603680ec7d3cbe8/cypress/integration/purpleAirMarker.spec.js#L46

So let's change it to just use a CSS class for /any/ sensor. You can load the map, visit some place, then inspect the HTML DOM and see if you can get a look at the structure of the HTML inside the `.leaflet-popup-content` element. Let's just change that one line to be more general - not tied to the specific name. 

Once we confirm in a PR that that specific error is gone, let's repeat in a new commit with another error. 

>       AssertionError: Timed out retrying: Too many elements found. Found '3', expected '2'.

This one expected 2 sensors. But it turned out there are 3 there now, because this is a live API and new people put new sensors online. So let's just change the test to assert/expect that there should be more than zero. 

How does that sound?

--- a/cypress/integration/owm.spec.js
+++ b/cypress/integration/owm.spec.js
@@ -6,7 +6,7 @@ describe('OpenWeatherMap layer', function() {
       cy.get('.leaflet-tile-pane').children().should('have.length', 1)
       cy.get('#map-Clouds label').click({ force: true })
       cy.get('.leaflet-tile-pane').children().should('have.length', 2)
-      cy.get('.img')
+      cy.get('img')
     }) 
   })
   

--- a/cypress/integration/owm.spec.js
+++ b/cypress/integration/owm.spec.js
@@ -4,7 +4,7 @@ describe('OpenWeatherMap layer', function() {
     cy.wait(300)
     cy.window().then((win) => {
       cy.get('.leaflet-tile-pane').children().should('have.length', 1)
-      cy.get('#map-city label').click({ force: true })
+      cy.get('#map-Clouds label').click({ force: true })
       cy.get('.leaflet-tile-pane').children().should('have.length', 2)
       cy.get('.img')
     }) 

--- a/cypress/integration/owm.spec.js
+++ b/cypress/integration/owm.spec.js
@@ -6,7 +6,7 @@ describe('OpenWeatherMap layer', function() {
       cy.get('.leaflet-tile-pane').children().should('have.length', 1)
       cy.get('#map-Clouds label').click({ force: true })
       cy.get('.leaflet-tile-pane').children().should('have.length', 2)
-      cy.get('img')
+      cy.get('img').should('have.attr', 'src').and('contains', 'openweathermap.org/map/clouds')
     }) 
   })
   

--- a/cypress/integration/owm.spec.js
+++ b/cypress/integration/owm.spec.js
@@ -6,7 +6,7 @@ describe('OpenWeatherMap layer', function() {
       cy.get('.leaflet-tile-pane').children().should('have.length', 1)
       cy.get('#map-Clouds label').click({ force: true })
       cy.get('.leaflet-tile-pane').children().should('have.length', 2)
-      cy.get('img').should('have.attr', 'src').and('contains', 'openweathermap.org/map/clouds')
+      cy.get('div.leaflet-layer:nth-child(2) > div > img').should('have.attr', 'src').and('contains', 'openweathermap.org/map/clouds')
     }) 
   })
   

--- a/cypress/integration/owm.spec.js
+++ b/cypress/integration/owm.spec.js
@@ -13,8 +13,13 @@ describe('OpenWeatherMap layer', function() {
   it('loads correct owmloading.gif with config option set', function() {
     cy.openWindow('/example/oneLinerCodeExample.html')
     cy.wait(300)
-    cy.window().then((win) => {
-      expect(LEL.overlayMaps.current.options.imageLoadingUrl).to.equal('https://raw.githubusercontent.com/buche/leaflet-openweathermap/master/owmloading.gif')
-    })
+    let LEL;
+    cy.window()
+      .then((win) => {
+        LEL = win.LEL
+      })
+      .then(() => {
+        expect(LEL.overlayMaps.current.options.imageLoadingUrl).to.equal('https://raw.githubusercontent.com/buche/leaflet-openweathermap/master/owmloading.gif')
+      })
   })
 });

--- a/cypress/integration/purpleAirMarker.spec.js
+++ b/cypress/integration/purpleAirMarker.spec.js
@@ -43,7 +43,7 @@ describe('Purple layer - markers', function() {
   it('shows popup', function() {
     cy.get('.leaflet-overlay-pane svg g').children().last().click({ force: true })
     cy.get('.leaflet-popup-pane').children().should('have.length', 1)
-    cy.get('.leaflet-popup-content').should('contain', 'MandMnorth40')
+    cy.get('.leaflet-popup-content').should('contain', 'www.purpleair.com')
   })
 
   it('removes markers from the map and the layer name from the hash when clicked again', function() {

--- a/cypress/integration/purpleAirMarker.spec.js
+++ b/cypress/integration/purpleAirMarker.spec.js
@@ -32,7 +32,7 @@ describe('Purple layer - markers', function() {
     const spy = cy.spy(window.top.aut.PurpleAirMarkerLayer, 'requestData')
     cy.get('[title="Show minimal markers"]').click().then(() => {
       expect(spy).to.be.called
-      cy.get('.leaflet-overlay-pane svg g').children().should('have.length', 2)
+      cy.get('.leaflet-overlay-pane svg g').children().should('have.length.gt', 0)
       cy.get('.leaflet-overlay-pane svg g path').invoke('attr', 'stroke').should('eq', '#7c7c7c')
       cy.get('.leaflet-overlay-pane svg g path').invoke('attr', 'fill').should('eq', '#7c22b5')
       cy.get('.leaflet-overlay-pane svg g path').invoke('attr', 'stroke-linecap').should('eq', 'round')

